src/graphcast_agent/message_typing.rs

```diff
@@ -403,6 +395,25 @@ mod tests {
         pub content: String,
     }
 
+    impl RadioPayload for SimpleMessage {
+        //TODO: Add various requirements to RadioPayload
+        // type ExternalValidation = Option<String>;
+        // fn validity_check(&self, _gc: GraphcastMessage<Self>, _val: Option<String>) -> Result<&Self, MessageError> {
+        //     Ok(self)
+        // }
+        fn valid_outer(&self, outer: &GraphcastMessage<Self>) -> Result<&Self, MessageError> {
```

Comment:
If this is something we expect all custom Radio message types to implement, how custom should/could the impl be? If we expect `valid_outer` to always be the same, shouldn't we enforce it on SDK side?

Reply:
We should require all Radios to check if the inner payload is consistent with the general Graphcast message, although I don't expect them to always be the same. For example `SimpleMessage` which doesn't contain fields like `nonce` even though it is more of an example message type. Since we don't have a restriction on the fields for the payload type, I don't think it will be easy to check the fields between the inner and outer types. 
Let me know if you think we should remove `valid_outer` altogether; maybe there's some other way to refactor the messages so the outer type doesn't have overlapping fields (currently `identifier, nonce, graph_account, etc` must be in inner for the signature and they are on the outer type for interacting with gossips and generic validations

Reply:
ah this makes total sense, in the current setup this is the best way then, thank you for explaining! 💯